source/casts/index.ts

```diff
@@ -55,14 +60,30 @@ export class Casts {
 			start = mapCoordinateToIndex(filter.geoFilter.location, 1);
 			end = start + 1;
 		}
-		return this.getSequenceQueryResult(this.sequenceQueryBuilder(level, filter.year), null, start, end, stream);
+		const promises = [];
+		if (years.length > 0) {
+			for (const year of years) {
+				for (const query of this.sequenceQueryBuilder(level, year, filter.provider)) {
+					promises.push(() => this.getSequenceQueryResult(query, null, start, end, stream));
+				}
+			}
+		} else {
+			for (const query of this.sequenceQueryBuilder(level, undefined, filter.provider)) {
+				promises.push(() => this.getSequenceQueryResult(query, null, start, end, stream));
+			}
+		}
+		const result = await throttleActions(promises, this._concurrency, stream);
+		const all = [];
+		for (const iterator of result) {
+			all.push(...iterator);
+		}
+		return all;
```

Comment:
I've seen this pattern a few times, but this is a good use-case for `flatMap`:
```
return result.flatMap((castCount) => castCount)
```

you can even simplify cases where you have to alter the content as well, for example:
```
const result = await Promise.all(promises);
for (const iterator of result) {
  const filtered = this.postCastFilter(iterator, filter);
  all.push(...filtered);
}
return all;
```
would just become:
```
return result.flatMap(iterator => this.postCastFilter(iterator, filter))
```

You'd have to upgrade the `lib` value in `tsconfig.json` from `es2018` to `es2019` for typescript to accept it.

Reply:
Nice, I wasn't aware of flatMap